src/components/AnalysisPage/UnitsSelector/UnitsSelector.jsx

```diff
@@ -7,26 +7,14 @@ import {
 import "./unitsSelector.css";
 import PropTypes from "prop-types";
 import { updatePartUnits } from "./updatePartUnits";
-import { getPartData } from "./getPartData";
 
 const UnitsSelector = ({ partId, units, setUnits }) => {
   const [showSuccess, setShowSuccess] = useState(false);
   //used to maintain the purple border around the select when it is active
   const [isActive, setIsActive] = useState(false);
 
   useEffect(() => {
-    const getCurrentUnits = async () => {
-      if (partId) {
-        try {
-          const data = await getPartData(partId);
-          setUnits(data.units);
-        } catch (error) {
-          console.error("Error getting part data:", error);
-        }
-      }
-    };
-
-    getCurrentUnits();
+    setUnits(null);
```

Comment:
I'm looking at commit dae2d2c28dc02738489bb99c62cb0fd2483af3dd

I actually think that the units should be populated with whatever units the part is set to in the database currently. Because for each part there is going to be a 'correct' units for that part because its whatever units the designer intended the part to be modeled in. So when a part is loaded in it should bet set to those units. The only real use case for someone doing an analysis with both types of units is just if they accidentally selected the wrong one first or are curious about the size in the other units or something but there is generally just going to be one correct units for the part and so after they set that type of units then the part should be set to that units again when it gets loaded back in.
So we shouldn't be setting the units always back to null when a part is loaded in BUT now that I'm reading this code again I also don't think that the unit selector should be making the HTTP request to get the part data and then set the units there. I'm pretty sure that the View parts modal is already making that call when a part is selected and so it should probably just save the units after that call and then the unit selector can get the units passed down to it and put that value in the unit selector when the part is loaded in.

Does that make sense?

Reply:
> Because for each part there is going to be a 'correct' units for that part because its whatever units the designer intended the part to be modeled in. So when a part is loaded in it should bet set to those units. The only real use case for someone doing an analysis with both types of units is just if they accidentally selected the wrong one first or are curious about the size in the other units or something but there is generally just going to be one correct units for the part and so after they set that type of units then the part should be set to that units again when it gets loaded back in.
> 

Ooooohh, see I didn't understand that this was the workflow. I kept thinking that it was annoying that it was just automatically set to a unit, not giving the person a fresh slate to choose from the beginning.

Reply:
viewpartsbutton is setting the units now!